rts/lra/coordinator/src/main/java/io/narayana/lra/coordinator/domain/model/LongRunningAction.java

```diff
@@ -770,8 +790,13 @@ public LRAParticipantRecord enlistParticipant(URI coordinatorUrl, String partici
                         compensatorData, version);
                 if (participant != null) {
                     // need to remember that there is a new participant
-                    deactivate(); // if it fails the superclass will have logged a warning
-                    savedIntentionList = true; // need this clean up if the LRA times out
+                    if (deactivate()) { // if it fails the superclass will have logged a warning
+                        savedIntentionList = true; // need this clean up if the LRA times out
+                    } else {
+                        throw new ServiceUnavailableException(LRALogger.i18nLogger.warn_saveState(DEACTIVATE_REASON));
+                    }
+                } else {
+                    throw new ServiceUnavailableException(LRALogger.i18nLogger.warn_saveState(DEACTIVATE_REASON));
```

Comment:
Please can you let me know why at this point detecting the participant was null (I presume, not enlisted) should be reported as inability to save state related to deactivation?

Reply:
Thank you!